src/components/Reserve.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { ToastContainer, toast, Slide } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.min.css';
+import { addReservation } from '../redux/slices/reservations';
+import { fetchDoctors } from '../redux/slices/doctors';
+import 'bulma/css/bulma.min.css';
+
+function CreateReservation() {
```

Comment:
- [ ] OPTIONAL, fair warning: When  I open a details doctor page and click the reserve button there:

![image](https://github.com/ilyassatlassi/final-capstone-project-frontend/assets/57421823/0d5100bc-7d88-481e-9290-190e1d8ba6a2)

I get to this page, with the Dr. Michael Johnson already selected:

![image](https://github.com/ilyassatlassi/final-capstone-project-frontend/assets/57421823/e1b0cfd7-a641-47c1-a277-49fbf8d032f6)

The issue is, that it's not actually selected because if I fill in the rest of the form and press "Reserve doctor":

![image](https://github.com/ilyassatlassi/final-capstone-project-frontend/assets/57421823/181103c5-655e-4887-bd82-5cf4271771c1)


No doctor id is sent to the server and thus the reservation fails.

You could either 1. Enforce the manual selection of a doctor by making the default option "Please select a doctor",  or 2.  Make the default selection come prefilled with the correct id.

Reply:
@Meltrust thanks for the review. I noticed the reservations fail on the first attempt. Thanks for pointing out the issue.